configure.py

```diff
@@ -1608,35 +1606,35 @@ def configure_static(o):
 
 
 def write(filename, data):
-  print_verbose('creating %s' % filename)
-  with open(filename, 'w+') as f:
+  print_verbose(f'creating {filename}')
+  with open(filename, 'w+', encoding='utf-8') as f:
     f.write(data)
 
 do_not_edit = '# Do not edit. Generated by the configure script.\n'
 
 def glob_to_var(dir_base, dir_sub, patch_dir):
-  list = []
-  dir_all = '%s/%s' % (dir_base, dir_sub)
+  file_list = []
+  dir_all = f'{dir_base}/{dir_sub}'
   files = os.walk(dir_all)
   for ent in files:
-    (path, dirs, files) = ent
+    (_, _1, files) = ent
     for file in files:
       if file.endswith(('.cpp', '.c', '.h')):
         # srcfile uses "slash" as dir separator as its output is consumed by gyp
-        srcfile = '%s/%s' % (dir_sub, file)
+        srcfile = f'{dir_sub}/{file}'
         if patch_dir:
-          patchfile = '%s/%s/%s' % (dir_base, patch_dir, file)
+          patchfile = f'{dir_base}/{patch_dir}/{file}'
           if os.path.isfile(patchfile):
-            srcfile = '%s/%s' % (patch_dir, file)
-            info('Using floating patch "%s" from "%s"' % (patchfile, dir_base))
-        list.append(srcfile)
+            srcfile = f'{patch_dir}/{file}'
+            info(f'Using floating patch "{patchfile}" from "{dir_base}"')
+        file_list.append(srcfile)
     break
-  return list
+  return file_list
 
 def configure_intl(o):
-  def icu_download(path):
+  def icu_download():
```

Comment:
I'm not sure about this change, the `path` of the `icu_download()` function was never used (not inside the function I mean), not even from the day it was added.

But an argument to the `path` parameter is being passed here:

https://github.com/nodejs/node/blob/069365c5bd63b8ed483a75e3791aae8f35ddf8c8/configure.py#L1819

even though it's not used inside the `icu_download()` function, is this a mistake or was it supposed to be used?

Reply:
Why should we revert this? Do you want me to make this change in a separate PR?